tensorflow/compiler/xla/service/gpu/nvptx_compiler.cc

```diff
@@ -234,6 +237,48 @@ bool MaybeLoadPtxFromFile(const HloModuleConfig module_config,
   return false;
 }
 
+std::unique_ptr<llvm::Module> MaybeLoadLLVMFromFile(const HloModule* module, llvm::Module* llvm_module) {
+  // If the xla_gpu_llvm_ir_file option is set, be explicit if a file is used
+  // and warn when a file is not used to ease catching typo in filename.
+  if (module == nullptr) {
+    return nullptr;
+  }
+
+  std::string prefix = xla::FilenameFor(*module, "", "");
+  std::string matched_filename;
+  for (const string& full_filename :
+       module->config().debug_options().xla_gpu_llvm_ir_file()) {
+    // To ease comparing many LLVM versions, accept different suffixes then
+    // the original filename.
+    auto filename = tensorflow::io::Basename(full_filename);
+    if (absl::StartsWith(filename, prefix)) {
+      matched_filename = full_filename;
+      VLOG(0) << "RunBackend() - Will load LLVM from file: " << full_filename;
+      break;
+    }
+  }
+  if (module->config().debug_options().xla_gpu_llvm_ir_file().size() > 0 &&
+      matched_filename.empty()) {
+    VLOG(0) << "RunBackend() - For module with prefix '" << prefix
+            << "', we did not found a LLVM file to load.";
+  }
+
+  if (!matched_filename.empty()) {
+    llvm::LLVMContext &context = llvm_module->getContext();
+    llvm::SMDiagnostic err;
+    std::unique_ptr<llvm::Module> loaded_module = llvm::parseIRFile(matched_filename, err, context);
+
+    if (!loaded_module) {
+      err.print("ERR", llvm::errs());
+      LOG(FATAL) << "Failed to load an LLVM file. It is probably invalid LLVM.";
```

Comment:
LOG(FATAL) seems a bit harsh here, maybe return StatusOr instead?

Reply:
This would change the function signature that is `std::unique_ptr<llvm::Module>`.
Here, we need to crash. I can change that function signature if you think this would give a better error message.
I do not know what between `StatusOr` and `LOG(FATAL)` give the best error message. Do you?